test/nyc-integration.js

```diff
@@ -1079,6 +1079,40 @@ describe('the nyc cli', function () {
     })
   })
 
+  it('help shows to stderr when main command doesn\'t know what to do', done => {
+    const msgs = {}
+    let waitingFor = 2
+    const doSpawn = (args, type, expectCode) => {
+      const proc = spawn(process.execPath, args, {
+        cwd: fixturesCLI,
+        env
+      })
+      msgs[type] = {
+        out: '',
+        err: ''
+      }
+      proc.stdout.on('data', chunk => {
+        msgs[type].out += chunk
+      })
+      proc.stderr.on('data', chunk => {
+        msgs[type].err += chunk
+      })
+      proc.on('close', code => {
+        code.should.equal(expectCode)
+        waitingFor--
```

Comment:
While I think this is clever, I feel like it could be done simpler. I've been wanting to introduce https://npm.im/await-spawn for a while, what do you think of it here?

Reply:
I think that would be cool once we can use async/await (nyc@15).  As long as we're still supporting node.js 6 and not transpiling test/*.js I'd rather leave it as is.

Reply:
It returns a promise so we don't have to use the new syntax just yet. Just:

```js
return spawn(...)
  .then(() => {
    //asserts
  })
```

Reply:
I'm OK with using await-spawn for these new tests but I don't think we should do a mass change to `return spawn(...).then(...)`.  I'd much rather we just leave existing tests alone until we can use async/await sytax.

Reply:
Actually it looks like `await-spawn` won't let us see both stdout and stderr.  I think our testing would be better served by something which mimics the API of `spawnSync`, just with promises.  So `spawn(...).then(({stdout, stderr, status}) => { ... })`.  This is something which is currently lacking IMO, we should be checking for empty stderr in most tests (including when the exit code is 0).

Reply:
Any feelings about using [execa](https://github.com/sindresorhus/execa#readme) in tests?  It's actually already installed via yargs so although it's a bigger module it wouldn't even increase our node_modules size.

Reply:
Sure! execa looks like a better fit, I agree :)

Reply:
Very nice.